commons-shared/src/main/scala/com/avsystem/commons/rpc/RPCTypeClasses.scala

```diff
@@ -8,13 +8,8 @@ package rpc
   * It's not required for RPC traits companion objects to extend this trait but it's recommended because it
   * reduces problems with incremental compilation and duplicated bytecode.
   */
-abstract class RPCTypeClasses[F <: RPCFramework with Singleton, T](implicit fullRPCMetadata: AllRPCTypeClasses[F, T]) {
-  implicit val asRawRPC: F#AsRawRPC[T] = fullRPCMetadata.asRawRPC
-  implicit val asRealRPC: F#AsRealRPC[T] = fullRPCMetadata.asRealRPC
-  implicit val rpcMetadata: F#RPCMetadata[T] = fullRPCMetadata.metadata
-}
-
-case class AllRPCTypeClasses[F <: RPCFramework with Singleton, T](asRealRPC: F#AsRealRPC[T], asRawRPC: F#AsRawRPC[T], metadata: F#RPCMetadata[T])
-object AllRPCTypeClasses {
-  implicit def materialize[F <: RPCFramework with Singleton, T]: AllRPCTypeClasses[F, T] = macro macros.rpc.RPCMacros.allRpcTypeClassesImpl[F, T]
-}
+abstract class RPCTypeClasses[F <: RPCFramework with Singleton, T] {
+  implicit def asRawRPC: F#AsRawRPC[T] = macro macros.rpc.RPCMacros.rpcTypeClassesImplAsRaw[F, T]
```

Comment:
How's that supposed to reduce code size if code now these instances get generated at _every_ callsite?
This is completely against what `RPCTypeClasses` was supposed to do - confine macro generated code to companion objects. This is not only about generated code size but also compilation speed and incremental compilation sanity.